bamboo-hr.sgnl.yaml

```diff
@@ -0,0 +1,99 @@
+displayName: "BambooHR"
+icon: # TODO: ask for nice icon
+description: "BambooHR as a System of Record"
+address: "api.bamboohr.com/api/gateway.php" 
+defaultSyncFrequency: HOURLY
+defaultSyncMinInterval: 1
+defaultApiCallFrequency: SECONDLY
+defaultApiCallMinInterval: 1
+type: "BambooHR-1.0.0"
+# Example Config: {"apiVersion":"v1","companyDomain":"sgnl","onlyCurrent":true,"attributeMappings":{"date":"yyyy-mm-dd","bool":{"true":["True","yes","1"],"false":["False","no","0"]}}}
+
+# Config Field Info:
+# [Required] ApiVersion: The version of the BambooHR API to use. Currently, only v1 is supported.
+
+# [Required] CompanyDomain: The subdomain used to access your account 
+# (If you access BambooHR at https://mycompany.bamboohr.com, then the subdomain is "mycompany")
+
+# [Optional] OnlyCurrent: If true, only the current employees will be returned.
+# If false, all employees will be returned. This defaults to false.
+
+# [Optional] AttributeMappings: A map of attribute types to potential mappings.
+# The BambooHR SoR Response returns all types as string values. By default, we will attempt to convert 
+# all attributes to the types specified in the catalog. This field allows you to specify additional mappings
+# to perform unconventional type conversions.
+
+# [Optional] AttributeMappings.Date: The format of the date string in the BambooHR response.
+# The BambooHR Admin Console allows you to configure date formats from four options.
+# To find this: Visit the BambooHR Admin Console > Settings > Account > General Settings > Date Input Format (Dropdown)
+# If this field is not specified, we default to "yyyy-mm-dd".
+
+# [Optional] AttributeMappings.Bool: A map of boolean values to their possible string representations.
+# The BambooHR bool type has many possible response values including "0/1", "yes/no", "true/false", etc.
+# This field allows you to specify the possible values for true and false.
+# By default, we will attempt a generic conversion for any attribute that is specified as a bool type in the catalog.
+# Golang recognizes these true/false combinations by default:
+# "1", "t", "T", "true", "TRUE", "True" are recognized as true.
+# "0", "f", "F", "false", "FALSE", "False" are recognized as false.
```

Comment:
Candidly, I really like this level of content here. It allows us to take it and reference in help.sgnl.ai in addition to what's in the template.

Reply:
My concern is these super helpful comments going unnoticed. 

A typical admin will see this, click on the SoR and land in SoR config page. If there's confusion in adapter-config, my instinct would be to check help.sgnl.ai instead of the GH repo. Plus, the GH repo is something the admin does not directly interact with at all throughout the Console usage lifecycle.


<img width="833" alt="image" src="https://github.com/SGNL-ai/catalog-sgnl-templates/assets/16449506/7f6721e7-81e6-4d36-a2db-afd4a3a03e8f">


As an improvement, we could 
- add a key called `docs` into catalog and put these comments as value
- make the FE show these docs when user hovers over a `?`/[info icon](https://www.iconfinder.com/icons/352432/info_icon)

WDYT @marc-sgnl?

Reply:
I'm ok to pull them out of the template, but I would like to see all supported filters as as well as all possible adapter config values documented to ensure they are used appropriately. If we can achieve that, I'm ok to forego the rest of this content. It's incredibly useful in building the help pages, so I'd like to see the time the team spent researching contributing to making our docs better.

Reply:
Let's keep it in the templates for now, and we can always move them or duplicate it in other places afterwards. WDYT @r-rakshith ?

Reply:
https://app.shortcut.com/sgnl/story/25002

Adding a feature request here for more dynamic and better documentation.